backend/app/repository/base_repository.py

```diff
@@ -13,16 +13,19 @@
 
 class BaseRepository(ABC, Generic[T]):
     _instance = None
+    _db = None
 
     def __new__(cls):
         if cls._instance is None:
             cls._instance = super(BaseRepository, cls).__new__(cls)
             try:
-                cls._instance.client = MongoClient(MONGODB_URI)
-                cls._instance.db = cls._instance.client.get_database('agh-thesis')
+                client = MongoClient(MONGODB_URI)
+                BaseRepository._db = client.get_database('agh-thesis')
             except ConnectionFailure as e:
                 raise DatabaseException(f'Cannot establish database, reason: {e}')
 
+        cls._instance.db = BaseRepository._db
+
```

Comment:
@AntuanW Nie wiem czy to sam wymyśliłeś czy to jest jakiś pattern, ale generalnie do tej pory chyba mieliśmy tyle połączeń do bazy ile różnych typów repository XD

`cls` to jest klasa dziedzicząca, więc do każdej z nich przypisywaliśmy nowe `_instance`

Ale kij z tym, `get_collection()` się wywalało (nie było używane przed tym PR)

Reply:
także to jest nadal nieco rozwalone (po co _instance w subklasie?) ale nie chcę zmieniać bez cb

Reply:
Tak, jest to wzorzec singleton po pythonowemu, bazujac na chyba Medium
Ale no nie bylo pod tym wzgledem przemyslane, to bardziej zabezpieczenie ze jak ktos pospami tym samym repo to sie stworzy jedno, na polaczenie z baza trzeba by miec jakies fixtures